File: Java/PrisonerSimulator/src/Player/Strategy/RandomStrategy.java
package Player.Strategy;

import Game.Decision;
import Player.IPlayerStrategyFacade;

import java.util.Random;

public class RandomStrategy implements IStrategy {
    private final Random random;

    public RandomStrategy() {
        random = new Random();
    }

    @Override
    public Decision MakeDecision(IPlayerStrategyFacade player, int rivalId) {
        boolean isCooperating = random.nextBoolean();
        return isCooperating ? Decision.Cooperate : Decision.Defect;
    }
}

File: Kotlin/PrisonerSimulator/src/player/strategy/RandomStrategy.kt
package player.strategy

import game.Decision
import player.IPlayerStrategyFacade
import kotlin.random.Random

class RandomStrategy : IStrategy {
    override fun makeDecision(player: IPlayerStrategyFacade, rivalId: Int): Decision =
        Random.nextBoolean().let {
            when (it) {
                true -> Decision.Cooperate
                false -> Decision.Defect
            }
        }

}
